src/FakeItEasy/Core/AssemblyExtensions.cs

```diff
@@ -14,7 +14,16 @@ public static bool ReferencesFakeItEasy(this Assembly assembly)
         {
             Guard.AgainstNull(assembly, nameof(assembly));
 
+#if FEATURE_REFLECTION_GETASSEMBLIES
             return assembly.GetReferencedAssemblies().Any(r => r.FullName == TypeCatalogue.FakeItEasyAssembly.FullName);
+#else
+            var fakeItEasyLibraryName = TypeCatalogue.FakeItEasyAssembly.GetName().Name;
+            var context = Microsoft.Extensions.DependencyModel.DependencyContext.Default;
+
+            var runtimeLib = context.RuntimeLibraries
+                .SingleOrDefault(library => string.Equals(library.Name, assembly.Name(), System.StringComparison.Ordinal));

```

Comment:
I notice that a library has a list of assemblies in it. Should we be looking at those?